src/Calculator.php

```diff
@@ -47,9 +48,10 @@ public function getDependencyInfo(string $directory, bool $verbose): array
 	 * @param Dependency $dependency
 	 * @param Repository[] $repositories
 	 * @param bool $verbose
+	 * @param string $directory
 	 * @return void
 	 */
-	private function updateVersionInfo(Dependency $dependency, array $repositories, bool $verbose)
+	private function updateVersionInfo(Dependency $dependency, array $repositories, bool $verbose, string $directory)
```

Comment:
Since the `$directory` doesn't change between dependencies, let's pull everything that gets the minimum stability out into its own method that can be called from `getDependencyInfo()`, and pass in the `$minStability` here instead